java/statestore/src/main/java/sleeper/statestore/StateStoreProvider.java

```diff
@@ -71,7 +77,17 @@ public StateStore getStateStore(TableProperties tableProperties) {
         return tableIdToStateStoreCache.get(tableId);
     }
 
-    public interface StateStoreLoader {
+    /**
+     * Creates an instance of the state store client for a Sleeper table. Implemented by {@link StateStoreFactory}.
+     */
+    public interface StateStoreCreator {
```

Comment:
I'm unsure about the name of this interface, we use StateStoreCreator as the name of a class that sets up the infrastructure for a state store. Perhaps this could be StateStoreClientProvider or StateStoreClientCreator?

Reply:
I've renamed it to just Factory. It's just a way to reference StateStoreFactory without using the concrete class that has AWS dependencies. I guess an alternative would be to rename StateStoreFactory to AwsStateStoreFactory, but we don't do that anywhere else.